scripts/generate-flow-types.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+PACKAGES=({packages/react,packages/jest})
+
+echo "Cleaning existing flow types from dist"
+find packages -type f -path '*/dist/*' -name '*.js.flow' -delete
+
+echo "Copying type overrides"
+find "${PACKAGES[@]}" -path '*/src/*' -name '*.js.flow' -not -path '*/node_modules/*' -not -path '*/__tests__/*' -print0 | while read -rd $'\0' override; do
+  echo "Found type override '$override'"
+  esmDirname=$(dirname "$override" | sed 's/\/src/\/dist\/esm/g')
+  if [ -d "${esmDirname%.*.*}" ]; then
+    cp "$override" "$(echo "$override" | sed 's/\/src/\/dist\/browser/g')"
+    cp "$override" "$(echo "$override" | sed 's/\/src/\/dist\/cjs/g')"
+    cp "$override" "$(echo "$override" | sed 's/\/src/\/dist\/esm/g')"
+  else
+    cp "$override" "$(echo "$override" | sed 's/\/src/\/dist/g')"
+  fi
+done
+
+echo "Generating remaining flow types with flowgen"
+find "${PACKAGES[@]}" -type f -path '*/dist/*' -name '*.d.ts' -print0 | while read -rd $'\0' file; do
+  if [ -f "${file%.*.*}.js.flow" ]; then
+    continue
+  fi
+
+  flowgen --add-flow-header "$file" -o "${file%.*.*}.js.flow"
+  if [ ! -f "${file%.*.*}.js.flow" ]; then
+    echo "Type '${file%.*.*}.js.flow' was not generated"
+    exit 1
+  fi
+done
+
+echo "Patching types"
+find "${PACKAGES[@]}" -type f -path '*/dist/*' -name '*.js.flow' -print0 | while read -rd $'\0' file; do
+  # Change import to import type (bug in flowgen)
+  sed -i '' 's/import {/import type {/g' "$file"
```

Comment:
Would like your opinion here @MonicaOlejniczak (and @madou if you're interested).

I put this together as a way to patch together the types as I corrected the errors. Do we think this is manageable enough for the dev loop in its current form? I'm not married to this idea but I liked how simple and readable it ended up so I kept the approach. 

I think plain transformation without parsing is sufficient but I could re-write this into TS.

Reply:
I think it's fine, since you should always be importing the type

In general, I prefer writing complex scripts with node / typescript, but it's fairly readable atm.